MekHQ/src/mekhq/gui/preferences/JWindowPreference.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019 The MegaMek Team. All rights reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.gui.preferences;
+
+import mekhq.preferences.PreferenceElement;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ComponentEvent;
+import java.awt.event.ComponentListener;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowStateListener;
+import java.lang.ref.WeakReference;
+
+public class JWindowPreference extends PreferenceElement implements WindowStateListener, ComponentListener {
+    private final WeakReference<Window> weakRef;
+    private int width;
+    private int height;
+    private int screenX;
+    private int screenY;
+    private boolean isMaximized;
+
+    public JWindowPreference(Window window) {
+        super (window.getName());
+
+        try {
+            this.width = window.getWidth();
+            this.height = window.getHeight();
+            this.screenX = window.getLocationOnScreen().x;
+            this.screenY = window.getLocationOnScreen().y;
+
+            if (window instanceof JFrame) {
+                this.isMaximized = (((JFrame)window).getState() & Frame.MAXIMIZED_BOTH) == Frame.MAXIMIZED_BOTH;
```

Comment:
Java docs [seem to indicate](https://docs.oracle.com/javase/7/docs/api/java/awt/Frame.html#getState()) you need to call [getExtendedState](https://docs.oracle.com/javase/7/docs/api/java/awt/Frame.html#getExtendedState()) to capture that possibility, but I'm not certain.

Reply:
Great catch!